tests/integration/testdata/invoke/event.json

```diff
@@ -1,4 +1,5 @@
 {
+    "utf8": "안녕",
```

Comment:
This works though, I would ask for a separate integ test for this. Reason being as a contributor, I want to know what my test actually broke. If this regresses, we will see some test fail with name "test_wait_for_result_no_error" but doesn't really tell me why or what I really broke at a glance. This is more of a raise the bar request as what you have works.

Reply:
The reason I kept it here is that UTF-8 support is inherently coupled to nearly all text representation (and more so for arbitrary user input) in our codebase, i.e. UTF-8 support is _the_ default case.

Reply:
Separated the tests; in the general case I don't think separating is necessary, but in this case it seems reasonable as there was a regression.